test/extended/deployments/deployments.go

```diff
@@ -210,6 +211,43 @@ var _ = g.Describe("deploymentconfigs", func() {
 			o.Expect(waitForLatestCondition(oc, "custom-deployment", deploymentRunTimeout, deploymentReachedCompletion)).NotTo(o.HaveOccurred())
 		})
 	})
+
+	g.Describe("with revision history limits", func() {
+		g.It("should never persist more old deployments than acceptable after being observed by the controller [Conformance]", func() {
+			revisionHistoryLimit := 3 // as specified in the fixture
+
+			_, err := oc.Run("create").Args("-f", historyLimitedDeploymentFixture).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			iterations := 10
+			for i := 0; i < iterations; i++ {
+				e2e.Logf("%02d: triggering a new deployment with config change", i)
+				out, err := oc.Run("set", "env").Args("dc/history-limit", fmt.Sprintf("A=%d", i)).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(out).To(o.ContainSubstring("updated"))
+
+				o.Expect(waitForLatestCondition(oc, "history-limit", deploymentRunTimeout, checkDeploymentConfigHasSynced)).NotTo(o.HaveOccurred())
```

Comment:
The first thing I would ensure here is that the ongoing deployment has completed and then that the dc is synced. Also go through the deploymentconfig controller and return c.updateStatus in every case the controller exits without an error (there are a couple of places)